system-design/app-interoperability/workload-orchestration-agent.md

```diff
@@ -83,9 +83,212 @@ The following shall be configurable to ensure compliance with local operations.
+```
+apiVersion: margo.org/v1-alpha1
+kind: ApplicationDeployment
+metadata:
+    id: 0f83e682-a35b-4533-ae93-ff3810095f31
+    applicationId: com-northstarida-digitron-orchestrator
+spec:
+    deploymentProfile:
+      type: helm.v3
+      components:
+        - name: digitron-orchestrator
+          properties:
+            repository: oci://northstarida.azurecr.io/charts/northstarida-digitron-orchestrator
+            revision: 1.0.9
+    parameters:
+      hostname:
+        value: edge.host.local
+        targets:
+          - pointer: edge_host_name
+            components: ["digitron-orchestrator"]
+          - pointer: global.ingress.host
+            components: ["digitron-orchestrator"]
+      pollFrequency:
+        value: 30
+        targets: 
+          - pointer: settings.pollFrequency
+            components: ["digitron-orchestrator"]
+      siteId:
+        value: d8b64e67-3bc4-4d0b-a3b8-da94b8f36315
+        targets:
+          - pointer: settings.siteId
+            components: ["digitron-orchestrator"]
+```
+
+**Top-level Attributes**
+
+| Attribute       | Type            | Required?       | Description     |
+|-----------------|-----------------|-----------------|-----------------|
+| apiVersion      | string    | Y    | Identifier of the version of the API the object definition follows.|
+| kind            | string    | Y    | Must be `ApplicationDeployment`.|
+| metadata        | Metadata    | Y    | Metadata element specifying characteristics about the application deployment. |
+| spec            | Spec    | Y    | Spec element that defines deployment profile and parameters associated with the application deployment. |
+
+**Metadata Attributes**
+
+| Attribute        | Type            | Required?       | Description     |
+|-----------------|-----------------|-----------------|-----------------|
+| id             | string          | Y    | The unique identifier UUID of the deployment specification. Needs to be assigned by the Workload Orchestration Software.|
+| applicationId         | string          | Y    | An identifier for the application. The id is used to help create unique identifiers where required, such as namespaces. The id must be lower case letters and numbers and MAY contain dashes. Uppercase letters, underscores and periods MUST NOT be used. The id MUST NOT be more than 200 characters. The applicationId MUST match the associated application package Metadata "id" attribute.|
+
+**Spec Attributes**
+
+| Attribute        | Type            | Required?       | Description     |
+|-----------------|-----------------|-----------------|-----------------|
+| deploymentProfile           | Deployment Profile          | Y    | Section that defines deployment details including type and components.|
+| parameters        | map[string][Parameter]          | Y    | Describes the configured parameters applied via the end-user.|
+
+**Deployment Profile Attributes**
```

Comment:
As we already defined this type in the "App Package" spec, I suggest to remove the definition here and instead put an anchor link into the description of "deploymentProfile" in the table above.

Same for the other types, inherited from "App Package".

Reply:
Agreed, removed and added link.